File: 0496-next-greater-element-i/0496-next-greater-element-i.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var nextGreaterElement = function(nums1, nums2) {
    let ans = [];
    for(let num of nums1){
        let gt = -1;
        for(let i = nums2.indexOf(num) + 1; i < nums2.length; i++){
            if(nums2[i] > num){
                gt = nums2[i];
                break;
            }
        }
        ans.push(gt);
    }
    return ans;
};

File: 0496-next-greater-element-i/0496-next-greater-element-i.rs
use std::collections::HashMap;
impl Solution {
    pub fn next_greater_element(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
        let mut stack = Vec::with_capacity(nums2.len());
        let mut map = HashMap::new();
        for n in nums2 {
            while !stack.is_empty() &&
            n > *stack.last().unwrap() {
                map.insert(stack.pop().unwrap(), n);
            }
            
            stack.push(n);
        }
        
        nums1
        .into_iter()
        .map(|n| *map.get(&n).unwrap_or(&-1))
        .collect()
    }
}